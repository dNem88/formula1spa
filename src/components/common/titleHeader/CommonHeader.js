import React from 'react'
import styles from './commonHeader.module.css'


function CommonHeader(props){

    return (
        <div className={styles['wrapper']}>
            <div className={styles.header}>
                <h1>{props.header}</h1>
            </div>
            <div className={styles.description}>
                <p >{props.description}</p>
            </div>
        </div>
    )
}

export default CommonHeader