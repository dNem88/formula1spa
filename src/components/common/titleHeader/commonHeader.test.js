import React from 'react'
import renderer from 'react-test-renderer'
import CommonHeader from './CommonHeader'
import {render, screen} from '@testing-library/react'


const mockProps = {
    header: 'test-header',
    description: 'test-description'
}

describe('Common Header Based on Props',() => {
    it('Match Snapshot', () => {
        const snapshot = renderer.create(
            <CommonHeader {...mockProps}/>
        ).toJSON()
        expect(snapshot).toMatchSnapshot()
    })
    it('Elements text content', () => {
        render(<CommonHeader {...mockProps}/>)
       expect(screen.getByText(mockProps.header)).toBeTruthy()
       expect(screen.getByText(mockProps.description)).toBeTruthy()
        
    })
})