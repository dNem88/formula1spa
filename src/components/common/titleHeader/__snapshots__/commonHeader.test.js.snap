// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Header Based on Props Match Snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="header"
  >
    <h1>
      test-header
    </h1>
  </div>
  <div
    className="description"
  >
    <p>
      test-description
    </p>
  </div>
</div>
`;
